python_modules/dagster/dagster/_utils/tags.py

```diff
@@ -0,0 +1,84 @@
+from collections import defaultdict
+from typing import Dict, Tuple
+
+from dagster import _check as check
+
+
+class TagConcurrencyLimitsCounter:
```

Comment:
unchanged other than method names to not mention runs

Reply:
might be nice to add something like `ITagged` so this all can be typed more accurately